libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -18,225 +22,166 @@
     this.context = context;
     this.accessToken = accessToken;
-    this.userAgent = userAgent;
-    this.client = client;
+    this.blacklistClient = new BlacklistClient(userAgent, new OkHttpClient(), this);
+    retrieveBlackList();
+  }
+
+  boolean isBlacklisted(String hash) {
+    return revokedKeys.contains(hash);
   }
 
-  List<String> retrieveBlackList() {
+  private void retrieveBlackList() {
     File directory = context.getFilesDir();
-    List<String> blacklist = new ArrayList<>();
+    revokedKeys = new ArrayList<>();
 
     if (directory.isDirectory()) {
       File file = new File(directory, BLACKLIST_FILE);
 
       if (file.exists()) {
         try {
-          blacklist = obtainBlacklistContents(file);
-          blacklist.remove(BLACKLIST_HEAD);
-        } catch (IOException | IndexOutOfBoundsException exception) {
+          if (!fileOpen) {
+            revokedKeys = obtainBlacklistContents(file);
+          }
+        } catch (IOException exception) {
           Log.e(LOG_TAG, exception.getMessage());
         }
       }
     }
-
-    return blacklist;
   }
 
   boolean daySinceLastUpdate() {
-    long millisecondDiff = System.currentTimeMillis() - retrieveLastUpdateTime();
-    return millisecondDiff >= DAY_IN_MILLIS;
-  }
-
-  private long retrieveLastUpdateTime() {
-    File directory = context.getFilesDir();
-    File file = new File(directory, BLACKLIST_FILE);
+    SharedPreferences sharedPreferences = TelemetryUtils.obtainSharedPreferences(context);
+    long lastUpdateTime = sharedPreferences.getLong(MAPBOX_SHARED_PREFERENCE_KEY_BLACKLIST_TIMESTAMP, 0);
 
-    long lastUpdateTime = 0;
-
-    if (file.exists()) {
-      try {
-        List<String> blacklist = obtainBlacklistContents(file);
-        lastUpdateTime = Long.valueOf(blacklist.get(BLACKLIST_HEAD));
-      } catch (IOException | IndexOutOfBoundsException exception) {
-        Log.e(LOG_TAG, exception.getMessage());
-      }
-    }
-
-    return lastUpdateTime;
+    long millisecondDiff = System.currentTimeMillis() - lastUpdateTime;
+    return millisecondDiff >= DAY_IN_MILLIS;
   }
 
-  void requestBlacklist(HttpUrl requestUrl) {
-    Request request = new Request.Builder()
-      .url(requestUrl)
-      .header(USER_AGENT_REQUEST_HEADER, userAgent)
-      .build();
-
-    client.newCall(request).enqueue(this);
+  void requestBlacklist() {
+    blacklistClient.requestBlacklist(blacklistClient.generateRequestUrl(context, accessToken));
   }
 
-  HttpUrl generateRequestUrl() {
-    return new HttpUrl.Builder().scheme(HTTPS_SCHEME)
-      .host(determineConfigEndpoint())
-      .addPathSegment(EVENT_CONFIG_SEGMENT)
-      .addQueryParameter(ACCESS_TOKEN_QUERY_PARAMETER, accessToken)
-      .build();
-  }
+  private void saveBlackList(ResponseBody responseBody) throws IOException {
+    fileOpen = true;
+    Gson gson = new GsonBuilder().create();
+    JsonObject responseJson = gson.fromJson(responseBody.string(), JsonObject.class);
```

Comment:
if `responseBody` is `null` this line will crash, if `responseBody.string()` returns empty string this will throw `JsonSyntaxException` which should be handled.